src/radical/pilot/agent/scheduler/spark.py

```diff
@@ -0,0 +1,97 @@
+
+__copyright__ = "Copyright 2013-2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import json
+import urllib2 as ul
+
+import radical.utils as ru
+
+from ... import utils     as rpu
+from ... import states    as rps
+from ... import constants as rpc
+
+from .base import AgentSchedulingComponent
+
+
+#===============================================================================
+#
+class Spark(AgentSchedulingComponent):
+
+    # FIXME: clarify what can be overloaded by Scheduler classes
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg):
+
+        AgentSchedulingComponent.__init__(self, cfg)
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        #-----------------------------------------------------------------------
+        # Find out how many applications you can submit to YARN. And also keep
+        # this check happened to update it accordingly
+
+
+        #if 'rm_ip' not in self._cfg['lrms_info']:
+        #    raise RuntimeError('rm_ip not in lm_info for %s' \
+        #            % (self.name))
+
+        self._log.info('Checking rm_ip %s' % self._cfg['lrms_info']['lm_info']['rm_ip'])
+        self._rm_ip = self._cfg['lrms_info']['lm_info']['rm_ip']
```

Comment:
A general remark: instead of

```
self._log.info('something with a %s' % string)
```

please use

```
self._log.info('something with a %s' % string)
```

In the first case the string will get expanded when the log call is made, so always when that code section is executed.  In the second case, it only gets expanded by the logging module when the log level requires it.  As string expansion can be rather expensive in python, we want the latter specifically in frequently called sections -- but it is a good habit to always use the comma'd version.